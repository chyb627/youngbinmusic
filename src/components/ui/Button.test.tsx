import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react-native';
import { Button } from './Button';
import { Typography } from './Typography';

test('Button Test', async () => {
  const onPressed = jest.fn();

  const expectedButtonName = 'TEST_BUTTON';

  render(
    <Button onPress={onPressed}>
      <Typography>{expectedButtonName}</Typography>
    </Button>,
  );

  expect(screen.toJSON()).toMatchSnapshot();

  fireEvent.press(screen.getByText(expectedButtonName));

  expect(onPressed).toHaveBeenCalled();
});

// 의도된 변화이면 업데이트 명령어를 다음과 같이 적어준다
// yarn test:unit -- --u
