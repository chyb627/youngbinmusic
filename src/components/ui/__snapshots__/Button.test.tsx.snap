// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Test 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={
    Object {
      "bottom": 0,
      "left": 0,
      "right": 0,
      "top": 0,
    }
  }
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": undefined,
      "paddingHorizontal": undefined,
      "paddingVertical": undefined,
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 10,
      }
    }
  >
    TEST_BUTTON
  </Text>
</View>
`;
